core/store/models/eth.go

```diff
@@ -260,6 +261,25 @@ type Head struct {
 	Number int64       `gorm:"index;not null"`
 }
 
+// AfterCreate is a gorm hook that trims heads after its creation
+func (h Head) AfterCreate(scope *gorm.Scope) (err error) {
+	scope.DB().Exec(`
+	DELETE FROM heads
+	WHERE id <= (
+	  SELECT id
+	  FROM (
+		SELECT id
```

Comment:
Could you check this with `pganalyze` to make sure that it is using indexes for this inner SELECT? There is a concern about the performance here the first time this query is run, when this table still has many rows we want to make sure that initial update runs as quickly as possible.

Reply:
![image](https://user-images.githubusercontent.com/40662484/62363133-1cde2180-b50e-11e9-9d3d-f803f26a81a1.png)
Is this what I should be looking at in pganalyze?